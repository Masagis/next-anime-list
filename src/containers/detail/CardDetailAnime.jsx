import React from 'react'

function CardDetailAnime(props) {
  const { data } = props
  return (
    <div>
      <div className="card shadow mb-3" data-testid="card-detail-anime">
        <div className="card-header text-center">Anime Detail</div>
        <div className="card-body">
          <div className="row">
            <div className="col-md-6">
              <table className="table table-hover">
                <tbody>
                  <tr>
                    <th>Synonyms</th>
                    <td data-testid="title-detail-anime">
                      {data?.title_japanese}
                    </td>
                  </tr>
                  <tr>
                    <th>Status</th>
                    <td data-testid="status-detail-anime">{data?.status}</td>
                  </tr>
                  <tr>
                    <th>Source</th>
                    <td data-testid="source-detail-anime">{data?.source}</td>
                  </tr>
                  <tr>
                    <th>Total Episode</th>
                    <td data-testid="episodes-detail-anime">
                      {data?.episodes}
                    </td>
                  </tr>
                  <tr>
                    <th>Producers</th>
                    <td data-testid="producer-detail-anime">
                      {data?.producers?.map((val) => `${val?.name}, `)}
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div className="col-md-6">
              <table className="table table-hover">
                <tbody>
                  <tr>
                    <th>English</th>
                    <td data-testid="english-detail-anime">
                      {data?.title_english}
                    </td>
                  </tr>
                  <tr>
                    <th>Tipe</th>
                    <td data-testid="type-detail-anime">{data?.type}</td>
                  </tr>
                  <tr>
                    <th>Durasi</th>
                    <td data-testid="duration-detail-anime">
                      {data?.duration}
                    </td>
                  </tr>
                  <tr>
                    <th>Studio</th>
                    <td data-testid="studios-detail-anime">
                      {data?.studios?.map((val) => `${val?.name}, `)}
                    </td>
                  </tr>
                  <tr>
                    <th>Rilis</th>
                    <td data-testid="rilis-detail-anime">
                      {data?.aired?.string}
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

export default CardDetailAnime
