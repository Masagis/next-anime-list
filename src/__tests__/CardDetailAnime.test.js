/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react'
import CardDetailAnime from '../containers/detail/CardDetailAnime'

describe('Component CardDetailAnime', () => {
  test('should be render the component card detail anime anime by id', () => {
    const data = {
      title_english: 'One Piece',
      title_japanese: 'ONE PIECE',
      type: 'TV',
      duration: '24 min',
      episodes: 12,
      aired: {
        from: '1999-10-20T00:00:00+00:00',
        to: null,
        prop: {
          from: {
            day: 20,
            month: 10,
            year: 1999,
          },
          to: {
            day: null,
            month: null,
            year: null,
          },
        },
        string: 'Oct 20, 1999 to ?',
      },
      source: 'Manga',
      status: 'Currently Airing',
      studios: [
        {
          mal_id: 18,
          type: 'anime',
          name: 'Toei Animation',
          url: 'https://myanimelist.net/anime/producer/18/Toei_Animation',
        },
      ],
      producers: [
        {
          mal_id: 169,
          type: 'anime',
          name: 'Fuji TV',
          url: 'https://myanimelist.net/anime/producer/169/Fuji_TV',
        },
        {
          mal_id: 416,
          type: 'anime',
          name: 'TAP',
          url: 'https://myanimelist.net/anime/producer/416/TAP',
        },
      ],
    }

    render(<CardDetailAnime data={data} />)
    expect(screen.getByTestId('card-detail-anime')).toBeInTheDocument()
    expect(screen.getByTestId('title-detail-anime')).toHaveTextContent(
      'ONE PIECE',
    )
    expect(screen.getByTestId('status-detail-anime')).toHaveTextContent(
      'Currently Airing',
    )
    expect(screen.getByTestId('source-detail-anime')).toHaveTextContent('Manga')
    expect(screen.getByTestId('episodes-detail-anime')).toHaveTextContent(12)
    expect(screen.getByTestId('producer-detail-anime')).toHaveTextContent(
      'Fuji TV',
    )
    expect(screen.getByTestId('english-detail-anime')).toHaveTextContent(
      'One Piece',
    )
    expect(screen.getByTestId('type-detail-anime')).toHaveTextContent('TV')
    expect(screen.getByTestId('duration-detail-anime')).toHaveTextContent(
      '24 min',
    )
    expect(screen.getByTestId('studios-detail-anime')).toHaveTextContent(
      'Toei Animation',
    )
    expect(screen.getByTestId('rilis-detail-anime')).toHaveTextContent(
      'Oct 20, 1999 to ?',
    )
  })

  test('snapshoot', () => {
    const data = {
      title_english: 'One Piece',
      title_japanese: 'ONE PIECE',
      type: 'TV',
      duration: '24 min',
      episodes: 12,
      aired: {
        from: '1999-10-20T00:00:00+00:00',
        to: null,
        prop: {
          from: {
            day: 20,
            month: 10,
            year: 1999,
          },
          to: {
            day: null,
            month: null,
            year: null,
          },
        },
        string: 'Oct 20, 1999 to ?',
      },
      source: 'Manga',
      status: 'Currently Airing',
      studios: [
        {
          mal_id: 18,
          type: 'anime',
          name: 'Toei Animation',
          url: 'https://myanimelist.net/anime/producer/18/Toei_Animation',
        },
      ],
      producers: [
        {
          mal_id: 169,
          type: 'anime',
          name: 'Fuji TV',
          url: 'https://myanimelist.net/anime/producer/169/Fuji_TV',
        },
        {
          mal_id: 416,
          type: 'anime',
          name: 'TAP',
          url: 'https://myanimelist.net/anime/producer/416/TAP',
        },
      ],
    }

    const renderComponent = render(<CardDetailAnime data={data} />)
    expect(renderComponent).toMatchSnapshot()
  })
})
