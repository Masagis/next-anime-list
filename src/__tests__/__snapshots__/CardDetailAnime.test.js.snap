// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CardDetailAnime snapshoot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="card shadow mb-3"
          data-testid="card-detail-anime"
        >
          <div
            class="card-header text-center"
          >
            Anime Detail
          </div>
          <div
            class="card-body"
          >
            <div
              class="row"
            >
              <div
                class="col-md-6"
              >
                <table
                  class="table table-hover"
                >
                  <tbody>
                    <tr>
                      <th>
                        Synonyms
                      </th>
                      <td
                        data-testid="title-detail-anime"
                      >
                        ONE PIECE
                      </td>
                    </tr>
                    <tr>
                      <th>
                        Status
                      </th>
                      <td
                        data-testid="status-detail-anime"
                      >
                        Currently Airing
                      </td>
                    </tr>
                    <tr>
                      <th>
                        Source
                      </th>
                      <td
                        data-testid="source-detail-anime"
                      >
                        Manga
                      </td>
                    </tr>
                    <tr>
                      <th>
                        Total Episode
                      </th>
                      <td
                        data-testid="episodes-detail-anime"
                      >
                        12
                      </td>
                    </tr>
                    <tr>
                      <th>
                        Producers
                      </th>
                      <td
                        data-testid="producer-detail-anime"
                      >
                        Fuji TV, 
                        TAP, 
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <div
                class="col-md-6"
              >
                <table
                  class="table table-hover"
                >
                  <tbody>
                    <tr>
                      <th>
                        English
                      </th>
                      <td
                        data-testid="english-detail-anime"
                      >
                        One Piece
                      </td>
                    </tr>
                    <tr>
                      <th>
                        Tipe
                      </th>
                      <td
                        data-testid="type-detail-anime"
                      >
                        TV
                      </td>
                    </tr>
                    <tr>
                      <th>
                        Durasi
                      </th>
                      <td
                        data-testid="duration-detail-anime"
                      >
                        24 min
                      </td>
                    </tr>
                    <tr>
                      <th>
                        Studio
                      </th>
                      <td
                        data-testid="studios-detail-anime"
                      >
                        Toei Animation, 
                      </td>
                    </tr>
                    <tr>
                      <th>
                        Rilis
                      </th>
                      <td
                        data-testid="rilis-detail-anime"
                      >
                        Oct 20, 1999 to ?
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="card shadow mb-3"
        data-testid="card-detail-anime"
      >
        <div
          class="card-header text-center"
        >
          Anime Detail
        </div>
        <div
          class="card-body"
        >
          <div
            class="row"
          >
            <div
              class="col-md-6"
            >
              <table
                class="table table-hover"
              >
                <tbody>
                  <tr>
                    <th>
                      Synonyms
                    </th>
                    <td
                      data-testid="title-detail-anime"
                    >
                      ONE PIECE
                    </td>
                  </tr>
                  <tr>
                    <th>
                      Status
                    </th>
                    <td
                      data-testid="status-detail-anime"
                    >
                      Currently Airing
                    </td>
                  </tr>
                  <tr>
                    <th>
                      Source
                    </th>
                    <td
                      data-testid="source-detail-anime"
                    >
                      Manga
                    </td>
                  </tr>
                  <tr>
                    <th>
                      Total Episode
                    </th>
                    <td
                      data-testid="episodes-detail-anime"
                    >
                      12
                    </td>
                  </tr>
                  <tr>
                    <th>
                      Producers
                    </th>
                    <td
                      data-testid="producer-detail-anime"
                    >
                      Fuji TV, 
                      TAP, 
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div
              class="col-md-6"
            >
              <table
                class="table table-hover"
              >
                <tbody>
                  <tr>
                    <th>
                      English
                    </th>
                    <td
                      data-testid="english-detail-anime"
                    >
                      One Piece
                    </td>
                  </tr>
                  <tr>
                    <th>
                      Tipe
                    </th>
                    <td
                      data-testid="type-detail-anime"
                    >
                      TV
                    </td>
                  </tr>
                  <tr>
                    <th>
                      Durasi
                    </th>
                    <td
                      data-testid="duration-detail-anime"
                    >
                      24 min
                    </td>
                  </tr>
                  <tr>
                    <th>
                      Studio
                    </th>
                    <td
                      data-testid="studios-detail-anime"
                    >
                      Toei Animation, 
                    </td>
                  </tr>
                  <tr>
                    <th>
                      Rilis
                    </th>
                    <td
                      data-testid="rilis-detail-anime"
                    >
                      Oct 20, 1999 to ?
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
